__tests__/src/components/ScrollTo.test.js

```diff
@@ -1,85 +1,65 @@
-
-        // It is called once when initially rendered w/ true
-        expect(scrollTo).toHaveBeenCalled();
-        scrollTo.mockReset();
-
-        rerender(<ScrollTo containerRef={containerRef}><div>Child</div></ScrollTo>);
 
-        // But it is not called on the re-render w/ false
-        expect(scrollTo).not.toHaveBeenCalled();
-      });
-    });
-
-    describe('when set from false to true', () => {
-      it('scrolls to the selected element when it is hidden above the container', () => {
-        jest.spyOn(ScrollTo.prototype, 'scrollToBoundingRect').mockImplementation(() => ({
-          ...scrollToElAboveBoundingRect,
-        }));
-        const { rerender } = render(<ScrollTo containerRef={containerRef}><div>Child</div></ScrollTo>);
-
-        rerender(<ScrollTo scrollTo containerRef={containerRef}><div>Child</div></ScrollTo>);
+    Element.prototype.scrollTo = jest.fn();
+  });
 
-        expect(scrollTo).toHaveBeenCalledWith(0, 230);
-      });
+  afterEach(() => {
+    Element.prototype.getBoundingClientRect = originalGetBoundingClientRect;
+    Element.prototype.scrollTo = originalScrollTo;
+    Element.prototype.offsetTop = originalOffsetTop;
+  });
 
-      it('scrolls to the selected element when it is hidden below the container', () => {
-        jest.spyOn(ScrollTo.prototype, 'scrollToBoundingRect').mockImplementation(() => ({
-          ...scrollToElBelowBoundingRect,
-        }));
+  it('renders the children', () => {
+    render(<ScrollTo><div data-testid="a" /></ScrollTo>);
 
-        const { rerender } = render(<ScrollTo containerRef={containerRef}><div>Child</div></ScrollTo>);
+    expect(screen.getByTestId('a')).toBeInTheDocument();
+  });
 
-        rerender(<ScrollTo scrollTo containerRef={containerRef}><div>Child</div></ScrollTo>);
+  it('scrolls to the element if it is off-screen ', () => {
+    const containerRef = createRef();
 
-        expect(scrollTo).toHaveBeenCalledWith(0, 230);
-      });
+    render(
+      <div data-testid="container" ref={containerRef} data-mockboundingrect={JSON.stringify({ bottom: 100, height: 100, top: 0 })}>
+        <div data-testid="scrollableContainer" style={{ height: 100, overflowY: true }} className="mirador-scrollto-scrollable">
+          <ScrollTo containerRef={containerRef}><div data-testid="a" style={{ height: 75 }} /></ScrollTo>
+          <ScrollTo containerRef={containerRef}><div data-testid="b" style={{ height: 75 }} /></ScrollTo>
+          <ScrollTo containerRef={containerRef} scrollTo><div data-testid="c" data-mockboundingrect={JSON.stringify({ bottom: 225, top: 150 })} style={{ height: 75 }} /></ScrollTo>
+        </div>
+      </div>,
+    );
 
-      it('does not scroll to the selected element when it is visible', () => {
-        jest.spyOn(ScrollTo.prototype, 'scrollToBoundingRect').mockImplementation(() => ({
-          ...visibleScrollToElBoundingRect,
-        }));
+    expect(Element.prototype.scrollTo).toHaveBeenCalled();
```

Comment:
Nice. Is there a way to test that it was actually called for `c`? Something like:
```
expect(Element.prototype.scrollTo).toHaveBeenCalledWith(0, -50);
```

Reply:
Because jest/jsdom doesn't implement offsetHeight, the calculated value is totally wrong so I don't think we want to write a test for it 🤷‍♂️ (and because it's a property, I don't think there's a ~good~ bad way to mock it out like I did the other methods).